AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/CreateKeys.dfy

```diff
@@ -308,37 +308,66 @@ module {:options "/functionSyntax:4" } CreateKeys {
     modifies storage.Modifies, kmsClient.Modifies
     ensures storage.ValidState() && kmsClient.ValidState()
 
-    //= aws-encryption-sdk-specification/framework/branch-key-store.md#versionkey
-    //= type=implication
-    //# VersionKey MUST first get the active version for the branch key from the keystore
-    //# by calling the configured [KeyStorage interface's](./key-store/key-storage.md#interface)
-    //# [GetEncryptedActiveBranchKey](./key-store/key-storage.md#getencryptedactivebranchkey)
-    //# using the `branch-key-id`.
-    ensures
-      && |storage.History.GetEncryptedActiveBranchKey| == |old(storage.History.GetEncryptedActiveBranchKey)| + 1
-      && Seq.Last(storage.History.GetEncryptedActiveBranchKey).input.Identifier == input.branchKeyIdentifier
-
-    ensures output.Success?
-            ==>
-              && Seq.Last(storage.History.GetEncryptedActiveBranchKey).output.Success?
-              && var oldActiveItem := Seq.Last(storage.History.GetEncryptedActiveBranchKey).output.value.Item;
-
+  {
+    var oldActiveItem :- fetchActiveItem(input.branchKeyIdentifier, storage, logicalKeyStoreName);
+    var hierarchyVersion := oldActiveItem.EncryptionContext[Structure.HIERARCHY_VERSION];
```

Comment:
FYI: I am dead certain the Storage layer ensures `hierarchy-version` is in the Branch Key Context.
This could be what is taking a long time to verify.